pkg/cpio/fs_unix.go

```diff
@@ -247,6 +247,13 @@ func (r *Recorder) GetRecord(path string) (Record, error) {
 		if done {
 			return Record{Info: info}, nil
 		}
+		// Check whether we can access it at all.
+		// If not, return an error. Callers of this function
+		// can determine whether that is fatal; in the cpio
+		// command, it should not be.
+		if err := unix.Access(path, unix.R_OK); err != nil {
```

Comment:
unix.Access does not do what you think it does.

Instead, try to open the file read-only and close it again

Reply:
Neither in this case does opening a file and closing it. To really test it, for some cases, you have to open a file, read a byte, and see if that works. That's because linux will succeed in some cases on everything but the read. I'm very reluctant to do that open/read test (although I did in an earlier version of this) b/c it's easy to cook up scenarios where that's not a very good idea (not necessarily on Linux, btw, i.e. where you want to package up a /proc as we do in Plan 9, a common scenario). Further, the standard cpio will create a  cpio record in the case or a read failing when the open succeeds, so calling that case a failing open is not right either. Among all these options (all of which I've tried), Access gives the most bang for the buck here, and treating a read fail as a short read elsewhere is the best option. That's what gives us behavior like standard cpio.

Reply:
Access doesn't even use the right UID/GID to check permissions, nor the right capabilities.

Why are we doing this here at all? Why not return an error when it happens? What's different between here and there?